### DISCUSSION TITLE
Welcome to the NeuralProphet Community!

### DISCUSSION THREAD

**Discussion by ourownstory** [ourownstory/neural_prophet#188] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other forecasting practitioners and NeuralProphet developers. We hope that you:
  * Ask any question.
  * Share ideas about how to improve NeuralProphet.
  * Engage with other community members and answer questions.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about how you use NeuralProphet.

**Comment by sumonsamanta123** [comment]
How do I use Neural Prophet when the period is in minutes. It seems to give me  error  'Train' object has no attribute 'lambda_delay'. I am trying to use Neural Prophet for forecasting live stock price. Timestamp of live stock price is in 5 minutes interval. So, the idea is sort the stock price from morning till 3:30 afternoon, and then predict the price for next 30 minutes. I understand that stock price theoretically follows  stochastic processes/Geometric Brownian Motion  - so may not get high level of accuracy using AR Net, but wanted to try. So, I would like to split the data till 3:30 and give it to neural prophet and then predict for next 30 minutes(and validate against the observed price.)

[intraday_5min_AMZN.txt](https://github.com/ourownstory/neural_prophet/files/6007617/intraday_5min_AMZN.txt)

**Reply by ourownstory** [reply]
Hi @sumonsamanta123 have a look at my response in https://github.com/ourownstory/neural_prophet/discussions/275

**Reply by sumonsamanta123** [reply]
Thank you, it works now with updates. But, forecast is not matching at all with observed data(not even the direction of up/down.) Real time series data has much more spike in either direction (AR Net forecast is in Red - for the region I was looking for). Just using ARMA (plus Grach, although did not change much with garch) gave better forecast - but not good enough. Will try multi variate LSTM(with many other related time series as lag term) and see.
![ts_plot](https://user-images.githubusercontent.com/79296883/108939124-362e4d80-761f-11eb-82b1-310c8fd63b59.png)

**Comment by luisgmc99** [comment]
Hi! I am currently using NeuralProphet for a project, I just updated from 0.2.7 to 0.5.3 version and did fixes in order to make it run, the thing that I am noticing is that when I am saving my model with joblib or pickle the file size is about 30GB while before it even stayed in KB, my question is, is this something that you guys had notice or am I doing something wrong? Thank you and I truly appreciate everything you've done.

**Reply by LeonieFreisinger** [reply]
Hey @luisgmc99, neuralprophet also offers a `.save()` method. I would recommend you to use that one.

**Reply by luisgmc99** [reply]
Oh, I tried but I got an error saying that the method did not exist, could you tell me where I could find documentation? Thanks a lot!

**Reply by LeonieFreisinger** [reply]
Sure, here you go: https://neuralprophet.com/code/utils.html?highlight=save#neuralprophet.utils.save

**Comment by luisgmc99** [comment]
I'm using NeuralProphet v0.2.7 and have trained a model on data from 2018 to 2022 to make predictions for 550 days. I saved the model and later received new data for the period from January 2023 to March 2023. I then loaded the saved model to make new predictions for another 550 days.

However, when comparing the new predictions with the previous ones, I noticed that the values for future regressors, season components, and the predictions themselves are exactly the same for overlapping dates, as if the model did not take into account the additional 3 months of data. Since my predictions are daily, I fed the model with approximately 90 new records.

I expected the predictions to change slightly with the new data. Is there a reason why the model doesn't seem to update its predictions when provided with new data? Am I missing a step or is this a limitation of the current implementation?

Any guidance would be much appreciated.

**Comment by adityamoglade** [comment]
File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py", line 213, in advance
    batch_output = self.batch_loop.run(kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/loop.py", line 199, in run
    self.advance(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/batch/training_batch_loop.py", line 90, in advance
    outputs = self.manual_loop.run(kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/loop.py", line 199, in run
    self.advance(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/optimization/manual_loop.py", line 110, in advance
    training_step_output = self.trainer._call_strategy_hook("training_step", *kwargs.values())
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/trainer/trainer.py", line 1494, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/strategies/strategy.py", line 378, in training_step
    return self.model.training_step(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/neuralprophet/time_net.py", line 791, in training_step
    self.log_dict(self.metrics_train(predicted_denorm, target_denorm), **self.log_args)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py", line 583, in log_dict
    self.log(
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py", line 422, in log
    apply_to_collection(
  File "/usr/local/lib/python3.10/dist-packages/lightning_utilities/core/apply_func.py", line 51, in apply_to_collection
    return function(data, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py", line 621, in __check_allowed
    raise ValueError(f"`self.log({name}, {value})` was called, but `{type(v)._name_}` values cannot be logged")
**ValueError: `self.log(RMSE, None)` was called, but `NoneType` values cannot be logged**



please how to solve this error in linux server

**Comment by jnbello** [comment]
Hi, I'm currently studying NeuralProphet (I have installed v. 0.6.2). Following examples in tutorials and books, 
This is my code:
model = NeuralProphet(changepoints_range=0.95, n_lags=10, ar_reg=1, num_hidden_layers=3, d_hidden=32)
metrics = model.fit(df_train, freq="B")

However,  i get the following error message:  "TypeError: __init__() got an unexpected keyword argument 'num_hidden_layers'"

Looking for information, I came to the following link: 
https://neuralprophet.com/how-to-guides/feature-guides/uncertainty_quantification.html

In section 2, "Conformal prediction", subsection "Base model training", I found a use of the "num_hidden_layers" parameter and the same error that I mentioned appears there.

What is happening? Radical changes from one version to another?

**Reply by ourownstory** [reply]
@leoniewgnr Can you please have a look at this?

**Reply by leoniewgnr** [reply]
@jnbello Yes, we changed it slightly. The new name is eg `ar_layers = [64, 64, 64, 64]` https://neuralprophet.com/code/forecaster.html

**Reply by leoniewgnr** [reply]
I'll fix the tutorial

**Reply by jnbello** [reply]
Thanks for your answer. I had come to that same conclusion after following the error within the forecaster.py code.
I also found another rule: ar_layers is a list of integers, where num_hidden_layers is the number of elements in that list, while d_hidden is the value in that same list.
Example, If num_hidden_layers=3 and d_hidden=32, then then ar_layers = [32, 32, 32]
Can you confirm the validity of this rule?

**Comment by ClanCultureDev** [comment]
> sort the stock price from morning till 3:30 afternoon, and then predict the price for next 30 minutes
